src/cython/sktda/preprocessing.py

```diff
@@ -0,0 +1,274 @@
+
+        Parameters:
+            X (list of n x 2 or n x 1 numpy arrays): input persistence diagrams.
+
+        Returns:
+            Xfit (list of n x 2 or n x 1 numpy arrays): thresholded persistence diagrams.
+        """
+        if self.use:
+            Xfit, num_diag = [], len(X)
+            for i in range(num_diag):
+                diag = X[i]
+                if self.point_type == "finite":
+                    if diag.shape[0] > 0:
+                        pers       = np.abs(np.matmul(diag[:,:2], [-1., 1.]))
+                        idx_thresh = pers >= self.threshold
+                        thresh_diag, thresh_pers  = diag[idx_thresh.flatten()], pers[idx_thresh.flatten()]
+                        sort_index  = np.flip(np.argsort(thresh_pers, axis=None), 0)
+                        if self.location == "upper":
+                            new_diag = thresh_diag[sort_index[:min(self.num_pts, thresh_diag.shape[0])],:]
+                        if self.location == "lower":
+                            new_diag = np.concatenate( [ thresh_diag[sort_index[min(self.num_pts, thresh_diag.shape[0]):],:], diag[~idx_thresh.flatten()] ], axis=0)
+                    else:
+                        new_diag = diag
+
+                else:
+                    if diag.shape[0] > 0:
+                        birth      = diag[:,:1]
+                        idx_thresh = birth >= self.threshold
+                        thresh_diag, thresh_birth  = diag[idx_thresh.flatten()], birth[idx_thresh.flatten()]
+                        if self.location == "upper":
+                            new_diag = thresh_diag[:min(self.num_pts, thresh_diag.shape[0]),:]
+                        if self.location == "lower":
+                            new_diag = np.concatenate( [ thresh_diag[min(self.num_pts, thresh_diag.shape[0]):,:], diag[~idx_thresh.flatten()] ], axis=0)
+                    else:
+                        new_diag = diag
+
+                Xfit.append(new_diag)                    
+        else:
+            Xfit = X
+        return Xfit
+
+class DiagramSelector(BaseEstimator, TransformerMixin):
+    """
+    This is a class for extracting finite or essential points in persistence diagrams.
+    """
+    def __init__(self, use=False, limit=np.inf, point_type="finite"):
+        """
+        Constructor for the DiagramSelector class.
+
+        Attributes:
+            use (bool): whether to use the class or not (default False).
+            limit (double): second coordinate value that is the criterion for being an essential point (default numpy.inf).
+            point_type (string): either "finite" or "essential". The type of the points that are going to be extracted.
```

Comment:
When there are infinite points, separating them is indeed one natural thing to do. Sometimes, people prefer instead to clamp values, i.e. replace any value larger than 42 (including infinity) with 42. Would that fit better as a Scaler, since it keeps the same number of points?

Reply:
Yes, this makes total sense. I added a `Clamping` class that can be called from within `DiagramScaler`